pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -817,25 +817,25 @@ void sock_dtls_init(void)
 
 static void _ep_to_session(const sock_udp_ep_t *ep, session_t *session)
 {
-    session->port = ep->port;
-    session->size = sizeof(ipv6_addr_t) +       /* addr */
-                    sizeof(unsigned short);     /* port */
+    dtls_session_init(session);
+    session->addr.port = ep->port;
+
     if (ipv6_addr_is_link_local((ipv6_addr_t *)ep->addr.ipv6)) {
         /* set ifindex for link-local addresses */
         session->ifindex = ep->netif;
     }
     else {
         session->ifindex = SOCK_ADDR_ANY_NETIF;
     }
-    memcpy(&session->addr, &ep->addr.ipv6, sizeof(ipv6_addr_t));
+    memcpy(&session->addr.addr, &ep->addr.ipv6, sizeof(ipv6_addr_t));
```

Comment:
```suggestion
    memcpy(&session->addr.addr, &ep->addr.ipv6, sizeof(session->addr.addr));
```
?

Reply:
Do you think it makes more sense to use the destination size rather than the source (I know they should equal)?

Reply:
My proposal uses the destination size? And yes, the destination size should always be used, as that buffer is the one at risk to be overloaded ;-).

Reply:
Changed